app/models/expression_of_interest.rb

```diff
@@ -0,0 +1,134 @@
+                :user_research_types,
+                :user_research,
+                :agree_privacy_statement,
+                :postcode,
+                :living_space,
+                :type,
+                :version,
+                :ip_address,
+                :user_agent,
+                :started_at,
+                :final_submission
+
+  validate :validate_different_address, if: -> { run_validation? :different_address }
+  validate :validate_accommodation_length, if: -> { run_validation? :accommodation_length }
+  validate :validate_more_properties, if: -> { run_validation? :more_properties }
+  validate :validate_number_adults, if: -> { run_validation? :number_adults }
+  validates :number_children, numericality: { only_integer: true, greater_than_or_equal_to: 0, less_than_or_equal_to: 9, message: I18n.t(:number_children, scope: :error) }, if: -> { run_validation? :number_children }
+  validate :validate_allow_pet_pet, if: -> { run_validation? :allow_pet }
+  validate :validate_user_research, if: -> { run_validation? :user_research }
+
+  after_initialize :after_initialize
+  before_save :serialize
+  before_save :generate_reference
+
+  after_find do
+    assign_attributes(answers)
+  end
+
+  def after_initialize
+    @family_types = %i[single_adult more_than_one_adult adults_with_children no_preference]
+    @accommodation_length_types = %i[from_6_to_9_months from_10_to_12_months more_than_12_months]
+    @final_submission = false
+    @different_address_types = %i[yes no]
+    @more_properties_types = %i[yes no]
+    @step_free_types = %i[all some none unknown]
+    @allow_pet_types = %i[yes no]
+    @user_research_types = %i[yes no]
+    @postcode = "not asked"
+    @living_space = "rooms_in_home_shared_facilities"
+  end
+
+  def as_json
+    {
+      id:,
+      reference:,
+      created_at:,
+      type:,
+      version:,
+      fullname:,
+      email:,
+      phone_number:,
+      residential_line_1:,
+      residential_line_2:,
+      residential_town:,
+      residential_postcode:,
+      different_address:,
+      property_one_line_1:,
+      property_one_line_2:,
+      property_one_town:,
+      property_one_postcode:,
+      more_properties:,
+      number_adults:,
+      number_children:,
+      family_type:,
+      accommodation_length:,
+      single_room_count:,
+      double_room_count:,
+      step_free:,
+      allow_pet:,
+      agree_future_contact:,
+      user_research:,
+      agree_privacy_statement:,
+      postcode:,
+      ip_address:,
+      user_agent:,
+      started_at:,
+    }.compact
+  end
+
+private
+
+  def validate_accommodation_length
+    validate_enum(@accommodation_length_types, @accommodation_length, :accommodation_length)
+  end
+
+  def serialize
+    self.type = "eoi-v2"
+    self.version = SCHEMA_VERSION
+    self.answers = as_json
+  end
+
+  def generate_reference
+    self.reference ||= sprintf("ANON-%<ref>s", ref: SecureRandom.uuid[9, 11].upcase)
```

Comment:
a couple of points here (unrelated to this specific work)

- we could rename `generate_reference` to something more meaningful for this piece
- will ask in Slack what's the rationale behind `ANON-...` and will suggest we rename it to `EOI-...`
- we should also bake the schema version into the app reference, for our own benefit if we have to look at the data

Reply:
I assumed it was just an arbitrary prefix, from derived 'anonymous'.

Reply:
updated to use EOI as prefix